diff --git a/pyproject.toml b/pyproject.toml
index 7ede6a2..818305e 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,7 +22,7 @@ gql = { extras = ["requests", "websockets"], version = "^3.3.0" }
 ujson = "^5.3.0"
 Deprecated = "^1.2.13"
 stringcase = "^1.2.0"
-attrs = "^23.1.0"
+attrs = ">23.1.0"
 httpx = "^0.25.0"
 
 [tool.poetry.group.dev.dependencies]
@@ -60,7 +60,6 @@ exclude = '''
 '''
 include = '\.pyi?$'
 line-length = 88
-target-version = ["py39", "py310", "py311", "py312", "py313"]
 
 [tool.commitizen]
 name = "cz_conventional_commits"
